### PR TITLE
HK: better error messaging for charm plando

### PR DESCRIPTION
**Pull request by qwint**
## What is this fixing or adding?
adds a specific error message for invalid plando charm costs

### CODE REVIEW

File: worlds/hk/Options.py [hunk 1]
```diff
@@ -300,7 +300,7 @@ class PlandoCharmCosts(OptionDict):
     display_name = "Charm Notch Cost Plando"
     valid_keys = frozenset(charm_names)
     schema = Schema({
-        Optional(name): And(int, lambda n: 6 >= n >= 0) for name in charm_names
+        Optional(name): And(int, lambda n: 6 >= n >= 0, error="Charm costs must be 0-6") for name in charm_names
```
Comment (Exempt-Medic) [inline worlds/hk/Options.py:303]:
```suggestion
        Optional(name): And(int, lambda n: 6 >= n >= 0, error="Charm costs must be in the range 0-6.") for name in charm_names
```

**Review by Exempt-Medic — APPROVED** [review]
Tested that it works. It can be a little funny that it gives the same error with a value of `2.5` but like... yeah that's fine

**Comment by Exempt-Medic** [comment]
> Tested that it works. It can be a little funny that it gives the same error with a value of `2.5` but like... yeah that's fine

I'm wondering if "[...] Charm costs must be integers in the range [...]" would be fine

**Comment by qwint** [comment]
feel free to suggest it and i'll merge it in lol, i'm not nitpicky about the text I just forgot that i had this as a proof of concept in my fork for a while so I PR'd it

File: worlds/hk/Options.py [hunk 2]
```diff
@@ -300,7 +300,7 @@ class PlandoCharmCosts(OptionDict):
     display_name = "Charm Notch Cost Plando"
     valid_keys = frozenset(charm_names)
     schema = Schema({
-        Optional(name): And(int, lambda n: 6 >= n >= 0) for name in charm_names
+        Optional(name): And(int, lambda n: 6 >= n >= 0, error="Charm costs must be in the range 0-6.") for name in charm_names
```
Comment (Exempt-Medic) [inline worlds/hk/Options.py:303]:
```suggestion
        Optional(name): And(int, lambda n: 6 >= n >= 0, error="Charm costs must be integers in the range 0-6.") for name in charm_names
```

**Review by Exempt-Medic — APPROVED** [review]
Reaffirming approval

**Review by Jouramie — APPROVED** [review]
Not tested. Code LGTM.

**Pull request merged**